### ISSUE TITLE
newrepo and toolchains

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxelib#378]
Build toolchains such has hxjava/hxcs does not work with the newrepo command.
Seems that they are still trying to find the toolchain from the global haxelib.

Example:
```
haxelib run hxcs hxcs_build.txt --haxe-version 3401 --feature-level 1
Error: Library hxcs is not installed
Error: Build failed
```

**Comment by kevinresol** [comment]
This is because the toolchain runs haxelib in the output folder, which does not contain the `.haxelib` folder

**Comment by kaikoga** [comment]
Setting ```HAXELIB_PATH=`pwd`/.haxelib``` explicitly for `haxelib run` worked for me.

Maybe `haxelib run` could `Sys.putEnv()` out of the box to actually allow toolchains with newrepo.

**Comment by RealyUniqueName** [comment]
This is fixed since #292. 
Now haxelib looks for `.haxelib` directory recursively up to the root of file system.
